q2_cutadapt/_demux.py

```diff
@@ -33,29 +34,61 @@ def run_command(cmd, verbose=True):
 
 
 def _build_demux_command(seqs_dir_fmt, barcode_fasta, per_sample_dir_fmt,
-                         untrimmed_dir_fmt, error_tolerance):
+                         untrimmed_dir_fmt, error_rate):
     cmd = ['cutadapt',
-           '-g', 'file:%s' % barcode_fasta.name,
-           '-e', str(error_tolerance),
-           # {name} is a cutadapt convention for interpolating the sample id
-           # into the filename.
-           '-o', '%s/{name}.fastq.gz' % str(per_sample_dir_fmt),
-           '--untrimmed-output',
-           '%s/forward.fastq.gz' % str(untrimmed_dir_fmt),
-           str(seqs_dir_fmt.file.view(FastqGzFormat)),
+           '--front', 'file:%s' % barcode_fasta.name,
+           '--error-rate', str(error_rate),
            ]
+    # PAIRED-END
+    if isinstance(seqs_dir_fmt, MultiplexedPairedEndBarcodeInSequenceDirFmt):
+        cmd = cmd + [
+            # {name} is a cutadapt convention for interpolating the sample id
+            # into the filename.
+            '-o', os.path.join(str(per_sample_dir_fmt), '{name}.1.fastq.gz'),
+            '-p', os.path.join(str(per_sample_dir_fmt), '{name}.2.fastq.gz'),
+            '--untrimmed-output',
+            os.path.join(str(untrimmed_dir_fmt), 'forward.fastq.gz'),
+            '--untrimmed-paired-output',
+            os.path.join(str(untrimmed_dir_fmt), 'reverse.fastq.gz'),
+            str(seqs_dir_fmt.forward_sequences.view(FastqGzFormat)),
+            str(seqs_dir_fmt.reverse_sequences.view(FastqGzFormat)),
+            ]
+    # SINGLE-END
+    else:
+        cmd = cmd + [
+            # {name} is a cutadapt convention for interpolating the sample id
+            # into the filename.
+            '-o', os.path.join(str(per_sample_dir_fmt), '{name}.fastq.gz'),
```

Comment:
If the name was always `{name}.1.fastq.gz` then you could share a bit more code between the single and paired-end cases (both in this helper, and in the rename).